nucleus/common/common-util/src/main/java/com/sun/common/util/logging/GFLogRecord.java

```diff
@@ -84,4 +84,25 @@ public String getThreadName() {
     public void setThreadName(String threadName) {
         this.threadName = threadName;
     }
+
+    /**
+     * CUSTOM-55
+     * in case of an object passed as a parameter, call it's toString() method
+     * to resolve it's values in the current thread, instead of waiting for queues / et
+     * so there is no possibility of state change of the object between thread
+     * @param params
+     * @return parameter array
+     */
+    private Object[] transform(Object[] params) {
+        if (params == null) {
+            return null;
+        }
+        for (int ii = 0; ii < params.length; ++ii) {
+            Object param = params[ii];
+            if (param != null && !param.getClass().equals(String.class)) {
+                params[ii] = params[ii].toString();
```

Comment:
For example UniformLogFormatter expects in some code paths java.util.Collection or java.util.Map. Now paremeters are just strings so it may require some adjustment of formatters. (or different approach)

Reply:
good catch, fixed